api/src/main/java/org/apache/iceberg/actions/ActionsProvider.java

```diff
@@ -35,6 +35,11 @@ default MigrateTable migrateTable(String tableIdent) {
         this.getClass().getName() + " does not implement migrateTable");
   }
 
+  default MigrateDeltaLakeTable migrateDeltaLakeTable(String tableIdent, String deltaS3Location) {
```

Comment:
can we also move classes in API and core to `iceberg-delta-lake`? It won't be a part of core `ActionsProvider` in API, but we can always use a mixin like `SupportsMigrateDeltaLake` for engine to implement new actions methods.

Reply:
Thank you for your suggestion. I moved everything relevant to `iceberg-delta-lake` and made the action api named `SupportMigrateDeltaLake`